src/participants/create_project.py

```diff
@@ -104,7 +104,11 @@ def get_repolinks(self, wid, project):
                 repolinks[repo].append(arch)
             if not repolinks[repo]:
                 del repolinks[repo]
-        return repolinks
+        if repolinks:
+            return repolinks
+        else:
+            # Creating a project with no repos makes no sense
+            raise RuntimeError("No suitable repos found in %s (must contain an arch in the name)" % project)
```

Comment:
if this was a generic create_project then yes
In fact I think it is really create_project_to_do_a_test_build
as such if it has nothing to build then it warrants an exception IMHO

I do agree that it should probably be raised around the:
if create:
from a logical perspective.